kotlin/query-features-with-arcade-expression/src/main/java/com/esri/arcgisruntime/sample/queryfeatureswitharcadeexpression/MainActivity.kt

```diff
@@ -0,0 +1,192 @@
+                    // get the screen point of the tapped map
+                    val screenPoint = Point(e.x.roundToInt(), e.y.toInt())
+                    // evaluate an Arcade expression on the point
+                    evaluateArcadeExpression(screenPoint)
+                    return true
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Evaluate the Arcade expression at the [tappedPoint] and use the result
+     * to show a callout with the crime in the last 60 days
+     */
+    private fun evaluateArcadeExpression(tappedPoint: Point) {
+        // get the layer based on the position tapped on the MapView.
+        val identifyLayerResultFuture = mapView.identifyLayerAsync(layer, tappedPoint, 12.0, false)
+        identifyLayerResultFuture.addDoneListener {
+            val identifyLayerResult = identifyLayerResultFuture.get()
+            // if layer is not identified, then return
+            if (identifyLayerResult == null || !identifyLayerResult.elements.any()) {
+                return@addDoneListener
+            }
+
+            // get the tapped GeoElement as an ArcGISFeature
+            val feature = identifyLayerResult.elements.first() as ArcGISFeature
+
+            // if the previously tapped feature is null or the previous feature ID does not match the current feature ID
+            // run the Arcade expression query to get the crime count for the tapped feature
+            if (previousFeature == null || (feature.attributes["ID"]?.equals(previousFeature?.attributes?.get("ID")) == false)) {
+                // show the loading indicator as the Arcade evaluator evaluation call can take time to complete
+                progressBar.visibility = View.VISIBLE
+                // instantiate a string containing the Arcade expression
+                val expressionValue =
+                    "var crimes = FeatureSetByName(\$map, 'Crime in the last 60 days');\n" +
+                        "return Count(Intersects(\$feature, crimes));"
+                // create an ArcadeExpression using the string expression
+                val arcadeExpression = ArcadeExpression(expressionValue)
+                // create an ArcadeEvaluator with the ArcadeExpression and an ArcadeProfile
+                val evaluator = ArcadeEvaluator(arcadeExpression, ArcadeProfile.FORM_CALCULATION)
+                // instantiate a list of profile variable key value pairs
+                val profileVariables = mapOf<String, Any>("\$feature" to feature, "\$map" to mapView.map)
+                // get the Arcade evaluation result future given the previously set profile variables
+                val resultFuture = evaluator.evaluateAsync(profileVariables)
+                resultFuture.addDoneListener {
+                    // get the result as an ArcadeEvaluationResult
+                    val arcadeEvaluationResult = resultFuture.get() as ArcadeEvaluationResult
```

Comment:
```suggestion
                    val arcadeEvaluationResult = resultFuture.get()
```

This cast is unnecessary I think